src/cirrus/cirrus_setup.py

```diff
@@ -0,0 +1,263 @@
+    apps = resp.json()
+    matched_app = None
+    for app in apps:
+        if app['app']['name'] == oauth_note:
+            matched_app = app
+            print "Token found for cirrus script... reusing it..."
+            break
+
+    if matched_app is None:
+        # need to create a new token
+        print "Creating a new Token for github access..."
+        resp = requests.post(
+            GITHUB_AUTH_URL,
+            auth=(user, passwd),
+            data=json.dumps({"scopes": ["gist", "repo"], "note": oauth_note})
+            )
+        resp.raise_for_status()
+        matched_app = resp.json()
+
+    token = matched_app['token']
+    url = matched_app['url']
+    result = {
+        'github-user': user,
+        'github-token': token,
+        'github-url': url,
+    }
+    del passwd
+    return result
+
+
+def request_pypi_credentials():
+    """
+    prompt the user to provide pypi credentials if not present
+
+    """
+    result = {'username': None, 'token': None}
+
+    user = ask_question(
+        'what is your pypi username?',
+        default=os.environ['USER']
+    )
+    token = getpass.getpass('what is your pypi token/password?')
+    result['username'] = user
+    result['token'] = token
+    return result
+
+
+def request_ssh_credentials():
+    """
+    prompt the user to provide ssh credentials if not present
+
+    """
+    result = {'username': None, 'keyfile': None}
+
+    user = ask_question(
+        'what is your ssh username?',
+        default=os.environ['USER']
+    )
+    keyfile = ask_question(
+        'what is your ssh key location?',
+        default=os.path.join(os.environ['HOME'], '.ssh', 'id_rsa')
+    )
+    result['username'] = user
+    result['keyfile'] = keyfile
+    return result
+
+
+def request_docker_credentials(auto_yes=False):
+    """
+    prompt the user for docker credentials
+
+    """
+    result = {'user': None, 'token': None, 'email': None}
+    use_docker = ask_question(
+        'Are you building and uploading docker images? [y/N]?',
+        default='n'
+        )
+    if 'n' in use_docker.lower():
+        return result
+    user = ask_question(
+        'what is your docker repo username?',
+        default=os.environ['USER']
+    )
+    result['user'] = user
+    email = ask_question(
+        'what is your docker repo email?',
+        default=None
+    )
+    token = getpass.getpass('what is your docker registry token/password?')
+    result['email'] = email
+    result['user'] = user
+    result['token'] = token
+    return result
+
+
+def request_buildserver_credentials():
+    """
+    prompt the user to see if a buildserver is being used, and if so
+    prompt the user to provide buildserver credentials if not present
+
+    """
+    result = {'user': None, 'token': None}
+
+    buildserver = ask_question(
+        'Are you triggering releases with a remote buildserver, eg Jenkins [y/N]?',
+        default='n'
+        )
+    if 'n' in buildserver.lower():
```

Comment:
I think folks do the caps switch for y/N on purpose, seems a little funny you switch it up. This would also accept 'Narwhale' as a negative response. Not blocking, just a little loose.